### PR TITLE
8 Add PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by alexiskelsall**
8 Add PATCH /api/articles/:article_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -219,3 +219,65 @@ describe.only("POST /api/articles/:article_id/comments", () => {
   })
 });
 
+describe.only("PATCH /api/articles/:article_id", () => {
+  test("200: Responds with the updated article", () => {
+    return request(app)
+      .patch("/api/articles/2")
+      .send({ inc_votes: 5 })
+      .expect(200)
+      .then(({ body: { updatedArticle } }) => {
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:228]: Since `updatedArticle` will only have one article in it, it is more conventional to pass it on the response body as a single object, rather than nested in an array. You can do this in your model (e.g. return `rows[0]`). That would also save you indexing for all your assertions.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -219,3 +219,65 @@ describe.only("POST /api/articles/:article_id/comments", () => {
   })
 });
 
+describe.only("PATCH /api/articles/:article_id", () => {
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:222]: Remove `.only` before merging to main

File: __tests__/app.test.js [hunk 3]
```diff
@@ -219,3 +219,65 @@ describe.only("POST /api/articles/:article_id/comments", () => {
...
+  test("200: Responds with the updated article", () => {
+    return request(app)
+      .patch("/api/articles/2")
+      .send({ inc_votes: 5 })
+      .expect(200)
+      .then(({ body: { updatedArticle } }) => {
+        expect(updatedArticle[0]).toHaveProperty("votes");
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:229]: Line 237 checks for this property. You don't need both assertions

File: __tests__/app.test.js [hunk 4]
```diff
@@ -219,3 +219,65 @@ describe.only("POST /api/articles/:article_id/comments", () => {
...
+      .send({ inc_votes: 4})
+      .then(()=>{
+       return request(app)
+        .patch("/api/articles/3")
+        .send({ inc_votes: -2})
+        .expect(200) 
+      })
+      .then(({ body: { updatedArticle } }) => {
+        expect(updatedArticle[0].votes).toBe(2)
+      });
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:252]: Nested `.then` 🚨 

This test will not be working as you expect it to. Rewrite it to just contain this promise chain, you also don't need to patch twice in order to test it works with negative numbers

File: __tests__/app.test.js [hunk 5]
```diff
@@ -219,3 +219,65 @@ describe.only("POST /api/articles/:article_id/comments", () => {
...
+  });
+  test("400: Responds with a 400 if number is not an interger", () => {
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:254]: Make test descriptions  precise. This `number` could be the `article_id` or the `inc_votes`

File: __tests__/app.test.js [hunk 6]
```diff
@@ -219,3 +219,65 @@ describe.only("POST /api/articles/:article_id/comments", () => {
...
+    return request(app)
+      .patch("/api/articles/2")
+      .send({ inc_votes: 4.5 })
+      .expect(400)
+      .then((res) => {
+        expect(res.body.error).toBe("Bad Request");
+      });
+
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:262]: Really nice to see this test 👍

File: __tests__/app.test.js [hunk 7]
```diff
@@ -219,3 +219,65 @@ describe.only("POST /api/articles/:article_id/comments", () => {
...
+});
+test("400: Responds with a 400 if nothing is sent", () => {
+  return request(app)
+    .patch("/api/articles/2")
+    .send({ })
+    .expect(400)
+    .then((res) => {
+      expect(res.body.error).toBe("Bad Request");
+    });
+});
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:281]: Since this endpoint is parametric (with a different model/controller), you also need to test that `article_id` is valid and is in the database.

File: db/Controllers/patch.controllers.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+const app = require("../app")
+const {updateArticleByID} = require("../Models/patch.models")
+
+function patchArticleByID (req, res, next){
+    if(!req.body || typeof req.body.inc_votes !== "number"){
+        return res.status(400).send({error: "Bad Request"})
+    }
+    
+    const votes = req.body
+    votes.article_id = req.params.article_id
```
Comment (nc-jo-watson) [inline db/Controllers/patch.controllers.js:10]: Missing `const` for this variable declaration. Also mutates `req.body`

Comment (nc-jo-watson) [reply db/Controllers/patch.controllers.js:10]: Line 9 assigns votes to `req.body`, not a copy of it.

File: db/Controllers/patch.controllers.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    updateArticleByID(votes)
```
Comment (nc-jo-watson) [inline db/Controllers/patch.controllers.js:11]: You could just pass `article_id` and `votes` in individually here, it would be clearer what you're trying to do.

**Review by nc-jo-watson — COMMENTED** [review]
Some good stuff here Lexi, but you're missing some tests and there is some code that needs correcting. See my comments.
